.travis.yml

```diff
@@ -264,6 +264,12 @@ os: linux
     cache:
       directories:
         - "$HOME/.ccache"
+        - depends/built
+        - depends/sources
+        - depends/work
+        - depends/x86_64-linux-gnu
+        - depends/x86_64-unknown-linux-gnu
+        - depends/x86_64-w64-mingw32
```

Comment:
Should we also add a line for MacOS? Looking at one of the last Travis builds, it seems to be using the folder `x86_64-apple-darwin17.7.0`.

And another question: does it make sense to temporarily enable the OSX build on this branch to check if these changes have a significant impact on its execution time?

Reply:
No. This caches directives are for linux builds only (and cross built windows images).
OSX builds rely on a totally different OS image hence the caching directives must be placed elsewhere and need a real env to determine where the dependencies builds get into play.

Reply:
I made a rapid test on a new branch based on this one and the MacOS cache seems to be working fine, the build time is reduced from [45](https://app.travis-ci.com/github/HorizenOfficial/zen/builds/256088249) to [15 minutes](https://app.travis-ci.com/github/HorizenOfficial/zen/builds/259004966).

Anyway, if you think it's not worth including such changes, feel free to ignore them, the PR is ok for me!

Reply:
Cherry picked